node/src/components/storage.rs

```diff
@@ -1578,24 +1530,33 @@ fn initialize_block_body_v1_db(
 ) -> Result<(), LmdbExtError> {
     info!("initializing v1 block body database");
     let mut txn = env.begin_rw_txn()?;
+
+    let block_body_hash_to_header_map =
+        construct_block_body_to_block_header_reverse_lookup(&txn, block_header_db)?;
+
     let mut cursor = txn.open_rw_cursor(*block_body_v1_db)?;
 
     for (raw_key, _raw_val) in cursor.iter() {
-        if deleted_block_body_hashes_raw.contains(raw_key) {
+        let key =
+            Digest::try_from(raw_key).map_err(|err| LmdbExtError::DataCorrupted(Box::new(err)))?;
+        if !block_body_hash_to_header_map.contains_key(&key) {
+            if !deleted_block_body_hashes_raw.contains(raw_key) {
+                // This means that the block body isn't referenced by any header, but no header
+                // referencing it was just deleted, either
+                warn!(?raw_key, "orphaned block body detected");
```

Comment:
In my opinion we should either:
* delete it, because we're confident it's safe
* error and shut down the node b/c we don't know what to do.

Reply:
I went for deleting it - note that this `if` doesn't skip the code afterwards.

Reply:
Ah, sorry. I missed that when initially commenting and then forgot to resolve the comment. Thanks.